src/cheri_insts.sail

```diff
@@ -705,26 +705,26 @@ union clause ast = CBuildCap : (regidx, regidx, regidx)
  *   operation.
  */
 function clause execute (CBuildCap(cd, cs1, cs2)) = {
-  let cs1_val = clearTagIfSealed(if unsigned(cs1) == 0 then DDC else C(cs1));
-  let cs2_val = C(cs2);
-  let targetCap = {unsealCap(cs2_val) with tag=cs1_val.tag};
-  let (cs1_base, cs1_top) = getCapBounds(cs1_val);
-  let (cs2_base, cs2_top) = getCapBounds(cs2_val);
-  let cs1_perms = getCapPerms(cs1_val);
-  let cs2_perms = getCapPerms(cs2_val);
-  let cs2_flags = getCapFlags(cs2_val);
-  let subset = (cs2_base >= cs1_base)
-             & (cs2_top <= cs1_top)
-             & (cs2_base <= cs2_top) /* check for length < 0 - possible because cs2 might be untagged */
-             & ((cs2_perms & cs1_perms) == cs2_perms);
-  let (exact, cd1) = setCapBounds(cs1_val, to_bits(cap_addr_width, cs2_base), to_bits(cap_len_width, cs2_top));
-  let (_, cd2) = setCapOffset(cd1, getCapOffsetBits(cs2_val)); // Ignore representability check, since Fast Rep Check not relevant
-  let cd3 = setCapPerms(cd2, cs2_perms);
-  let cd4 = setCapFlags(cd3, cs2_flags);
-  let derivable = cd4 == targetCap;
-  assert(not(derivable) | exact, "CBuildCap: setCapBounds was not exact"); /* base and top came from cs2 originally so will be exact */
-  let cd5 = if signed(cs2_val.otype) == otype_sentry then sealCap(cd4, to_bits(cap_otype_width, otype_sentry)) else cd4;
-  let cd6 = if subset & derivable then cd5 else clearTag(targetCap);
+  let authority_val = clearTagIfSealed(if unsigned(cs1) == 0 then DDC else C(cs1));
+  let requested_val = {C(cs2) with tag=authority_val.tag};
```

Comment:
This allows you to `CBuildCap` an already tagged cap. It might clear the tag if the `CBuildCap` would fail. Seems harmless to me, but I don't know if others have thoughts.

Reply:
That does seem a little strange, though it's hard to see a case where we'd knowingly be passing a tagged cap as the bag of bits to `CBuildCap` and possibly clearing the tag doesn't seem like it will make such an unforeseen situation any worse than it already would be.

That said, if we're changing things, might we prefer shifting to `CCBuildCap`-esque behaviour where tagged things are just passed through unmodified (for non-branching reconstruction with multiple roots).